tenant/pkg/controller/tenantnamespace/tenantnamespace_controller.go

```diff
@@ -50,16 +50,16 @@ const (
 // Add creates a new TenantNamespace Controller and adds it to the Manager with default RBAC. The Manager will set fields on the Controller
 // and Start it when the Manager is Started.
 func Add(mgr manager.Manager) error {
-	return add(mgr, newReconciler(mgr))
+	return AddManager(mgr, NewReconciler(mgr))
 }
 
-// newReconciler returns a new reconcile.Reconciler
-func newReconciler(mgr manager.Manager) reconcile.Reconciler {
+// NewReconciler returns a new reconcile.Reconciler
+func NewReconciler(mgr manager.Manager) reconcile.Reconciler {
 	return &ReconcileTenantNamespace{Client: mgr.GetClient(), scheme: mgr.GetScheme()}
 }
 
-// add adds a new Controller to mgr with r as the reconcile.Reconciler
-func add(mgr manager.Manager, r reconcile.Reconciler) error {
+// AddManager adds a new Controller to mgr with r as the reconcile.Reconciler
+func AddManager(mgr manager.Manager, r reconcile.Reconciler) error {
```

Comment:
If Add is exported, why do you need to export "AddManager"?

Reply:
We cannot pass the reconciler through the exported [Add](https://github.com/kubernetes-sigs/multi-tenancy/blob/a83c30e611b57fce779bf1262d036446c962a28b/tenant/pkg/controller/tenantnamespace/tenantnamespace_controller.go#L52) in the tenantnamespace controller, needed something similar to the [AddManger](https://github.com/kubernetes-sigs/multi-tenancy/blob/a83c30e611b57fce779bf1262d036446c962a28b/tenant/pkg/controller/tenant/tenant_controller_suite.go#L56) in the tenant controller.